compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -3029,17 +3046,22 @@ public void visit(BLangIntRangeExpression rangeExpr) {
         genNode(endExpr, env);
         rangeExpr.regIndex = calcAndGetExprRegIndex(rangeExpr);
 
+        RegIndex startExprRegIndex = startExpr.regIndex;
+        RegIndex endExprRegIndex = endExpr.regIndex;
+
         if (!rangeExpr.includeStart || !rangeExpr.includeEnd) {
             RegIndex const1RegIndex = getRegIndex(TypeTags.INT);
             emit(InstructionCodes.ICONST_1, const1RegIndex);
             if (!rangeExpr.includeStart) {
-                emit(InstructionCodes.IADD, startExpr.regIndex, const1RegIndex, startExpr.regIndex);
+                startExprRegIndex = getRegIndex(TypeTags.INT);
```

Comment:
Can we improve this old logic as well, by moving this to desugar as part of the above change?